### PR TITLE
feat(whiskers): add --create-parent-dirs option

### PR DESCRIPTION
**Pull request by backwardspy**

### CODE REVIEW

**Comment by nekowinston** [comment]
FWIW, I would expect this to be the default. I don't have to tell most build systems to generate the dist directory if I haven't done so manually.

As an aside, you probably want to keep using Conventional Commit PR titles, if you want PRs like this and #179 to show up in #174 automatically.

**Comment by backwardspy** [comment]
> FWIW, I would expect this to be the default. I don't have to tell most build systems to generate the dist directory if I haven't done so manually.

i've been back and forth on this a bit, i agree for build tools it makes sense for this to just happen. i'll think about it some more.

> As an aside, you probably want to keep using Conventional Commit PR titles, if you want PRs like this and #179 to show up in #174 automatically.

is that done based on PR titles? i assumed it would look at commits instead. will change them now!

**Comment by nekowinston** [comment]
> is that done based on PR titles? i assumed it would look at commits instead. will change them now!

https://github.com/catppuccin/toolbox/actions/runs/8858928452/job/24328133362#step:2:127

It's looking at the PRs, both at the title and text content, since you can override them retroactively with
```
BEGIN_COMMIT_OVERRIDE
type(scope): message
END_COMMIT_OVERRIDE
```

You should be able to just add that into #179 and it will show up in the changelog the next time release-please runs on main.

**Comment by backwardspy** [comment]
we discussed this a little further and concluded that it does make sense to just do this by default, especially as this change to _create_ directories is pretty non-destructive. overwriting existing files would possibly be a different discussion, but we're already doing that without an associated flag.

i've removed the cli flag entirely as i can't think of a reason to disable this behaviour. if one comes up, i think it should be rolled into the same change as the aforementioned overwriting of files.

File: whiskers/README.md [hunk 1]
```diff
@@ -55,6 +55,11 @@ Options:
       --overrides <OVERRIDES>
           Set frontmatter overrides
 
+      --create-parent-dirs
+          Automatically create parent directories before writing output
```
[the remark this answers is not in the document]
Comment (backwardspy) [reply whiskers/README.md]: resolved by removing the flag entirely 😎

**Comment by backwardspy** [comment]
thanks for the review & discussion!

**Pull request merged**